### PR TITLE
#162553610 Enable facebook login and signup

### PR DESCRIPTION
**Pull request by augustineezinwa**
#### What does this PR do?
- This PR enables facebook login/signup 
#### Description of Task to be completed?
-  enable redirect on successful signup
- authenticate user
- send in-app notifications  on successful facebook authentification
- write  actions and reducers
#### How should this be manually tested?
- `npm test`
#### Any background context you want to provide?
- Work in progress
#### What are the relevant pivotal tracker stories?
- #62553610
#### Screenshots (if appropriate)
- N/A
#### Questions:
- N/a
#### Link to review app:
  [Review App](https://valinor-ah-frontend-stag-pr-30.herokuapp.com)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const express = require('express');
 const path = require('path');
+require('dotenv').config();
```
Comment (darthrighteous) [inline index.js]: this dotenv initialization does not need to be here, you've already done this in the webpack config

File: .eslintrc.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
       "eol-last": 2,
       "no-unused-vars": 2,
       "no-underscore-dangle": 0,
+      "import/no-named-as-default":0,
```
Comment (darthrighteous) [inline .eslintrc.json]: why are we turning off this rule?

Comment (augustineezinwa) [reply .eslintrc.json]: @Darthrighteous resolved.

File: index.js (hunk 1 above)
Comment (augustineezinwa) [reply index.js]: @Darthrighteous done

File: src/components/common/FacebookButton.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import PropTypes from 'prop-types';
+
+const FacebookButton = ({ responseFacebook, request }) => (
+  <FacebookLogin
+  appId={process.env.FACEBOOK_APP_ID}
+  autoLoad={false}
+  onClick={request}
+  fields="name,email,picture"
+  callback={responseFacebook}
+  // eslint-disable-next-line react/jsx-no-bind
```
Comment (darthrighteous) [inline src/components/common/FacebookButton.js:12]: it doesn't look like you need this line anymore

File: src/index.html [hunk 4]
```diff
@@ -5,11 +5,13 @@
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
   <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
+  <link href="https://cdnjs.cloudflare.com/ajax/libs/toastr.js/latest/css/toastr.min.css" rel="stylesheet"/>
```
Comment (darthrighteous) [inline src/index.html]: is this still necessary?

File: src/index.html [hunk 5]
```diff
@@ -5,11 +5,13 @@
...
   <title>Author's Haven</title>
 </head>
 <body>
   <div id="app" class="main-container"></div>
   <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
   <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha384-ChfqqxuZUCnJSK3+MXmPNIyE6ZbWh2IMqE241rYiqJxyMiZ6OW/JmZQ5stwEULTy" crossorigin="anonymous"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/toastr.js/latest/toastr.min.js"></script>
```
Comment (darthrighteous) [inline src/index.html]: is this still necessary?

File: src/App.js [hunk 6]
```diff
@@ -24,8 +24,8 @@ class App extends Component {
 
   render() {
     return (
-      <PersistGate persistor={persistor}>
-        <Provider store={store}>
+      <Provider store={store}>
+        <PersistGate persistor={persistor}>
```
Comment (darthrighteous) [inline src/App.js:28]: was the original ordering incorrect?

**Review by darthrighteous — COMMENTED** [review]
Good job Augustine. please review my comments

File: src/App.js (hunk 6 above)
Comment (augustineezinwa) [reply src/App.js:28]: Yes, as regards to what is contained in the redux persist documentation

File: src/index.html (hunk 5 above)
Comment (augustineezinwa) [reply src/index.html]: done

**Review by sulenchy — APPROVED** [review]
LGTM

**Pull request merged**